src/traces/scattersmith/attributes.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+var hovertemplateAttrs = require('../../plots/template_attributes').hovertemplateAttrs;
+var texttemplateAttrs = require('../../plots/template_attributes').texttemplateAttrs;
+var extendFlat = require('../../lib/extend').extendFlat;
+var scatterAttrs = require('../scatter/attributes');
+var lineAttrs = scatterAttrs.line;
+
+module.exports = {
+    mode: scatterAttrs.mode,
+
+    re: {
+        valType: 'data_array',
+        editType: 'calc+clearAxisTypes',
+        description: 'Sets the real coordinates'
+    },
+
+    im: {
+        valType: 'data_array',
+        editType: 'calc+clearAxisTypes',
+        description: 'Sets the imaginary coordinates'
+    },
+
+    text: scatterAttrs.text,
+    texttemplate: texttemplateAttrs({editType: 'plot'}, {
+        keys: ['re', 'im', 'text']
```

Comment:
Please add tests for these `texttemplate` options.

Reply:
I am not sure how `texttemplate` works (even in other trace types that support it, like `scatterpolar`). In our initial design discussions for this plot, we deliberately stayed away from features like labels and zoom, etc., so I wonder if we could back this out now and add it in later, after a basic version of the plot is working.

Reply:
5a329dadaa3c89bceb8a7f2d38d1110cf4958ca5